File: solutions/c/triangle/1/triangle.c
#include "triangle.h"

bool is_degen(triangle_t s){
    return s.a + s.b < s.c || s.a + s.c < s.b || s.c + s.b < s.a || s.a <= 0 || s.b <= 0 || s.c <= 0;
}

bool is_equilateral(triangle_t sides){
    return sides.a == sides.b && sides.a == sides.c && !is_degen(sides);
}

bool is_isosceles(triangle_t sides){
    return (sides.a == sides.b || sides.a == sides.c || sides.b == sides.c) && !is_degen(sides);
}

bool is_scalene(triangle_t sides){
    return !is_isosceles(sides) && !is_degen(sides);
}

File: solutions/julia/triangle/1/triangle.jl
is_degen(sides) = minimum(sides) <= 0 || sum(sides) < 2 * maximum(sides)

is_equilateral(sides) = length(Set(sides)) == 1 && !is_degen(sides)

is_isosceles(sides) = length(Set(sides)) <= 2 && !is_degen(sides)

is_scalene(sides) = length(Set(sides)) == 3 && !is_degen(sides)